### DISCUSSION TITLE
I need custom background color for TextColumn and Select::make

### DISCUSSION THREAD

**Discussion by sergeynilov** [filamentphp/filament#6902] [Help] [answered]
In laravel 9 app with  filament 2.17.49 I have demands table with priority_level field(in migrations defined as enum with 6 values)
and I need depending on value in priority_level field to show different background color both in table :

```
     TextColumn::make('priority_level')->sortable()->searchable()
     ...
```

and in  the form :
```
     Select::make('priority_level')
         ->preload()
         ->options(Ticket::getPriorityLevelOptions())
         ->required(),

```
priority_level Keys=> colors are written in config file. I mean different colors, not standard danger, primary, success, warning...

How can I apply these colors to background color of TextColumn and Select controls?

**Comment by pboivin** [comment, marked as answer]
Something like this should work on the TextColumn (not sure about SelectColumn):

```php
    ->extraAttributes(function ($record) {
        $backgroundColor = match (true) {
            $record->priority_level > 90 => '#ff0000',
            // ...
            default => '#00ff00',
        };

        return ['style' => "background-color: $backgroundColor"];
    }),
```
